test/viewmodels/map_view_model_unit_test.dart

```diff
@@ -0,0 +1,57 @@
+import "package:flutter_test/flutter_test.dart";
+import "package:google_maps_flutter/google_maps_flutter.dart";
+import "package:mockito/mockito.dart";
+import "package:proxima/viewmodels/map_view_model.dart";
+
+class MockGoogleMapController extends Mock implements GoogleMapController {
+  @override
+  Future<void> animateCamera(CameraUpdate? cameraUpdate) {
+    return Future.value();
+  }
```

Comment:
This does not correspond to the boilerplate used for defining Mockito mocks. You look [here](https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md#manually-override-a-method-with-a-non-nullable-return-type) or at `MockUserRepositoryService` for examples.

If you don't need the features of `Mock` in your tests, I suggest you don't extend it.

Reply:
This mock is just here because it's not possible to create an Instance of GoogleMapController.